eng/templates/steps/run-build-from-source.yml

```diff
@@ -0,0 +1,53 @@
+parameters:
+  # Specifies whether to add the ArcadeBuildNonPortable flag to the build
+  includeNonPortable: 'false'
+
+  # Specifies the targetRID to pass to the build.
+  targetRID: ''
+
+steps:
+- script: |
+    set -x
+    df -h
+
+    # Setup build arguments
+    _BuildConfig="Release"
+    _InternalBuildArgs=""
+    if [[ "$BUILD_REASON" != "PullRequest" && "$SYSTEM_TEAMPROJECT" != "public" ]]; then
+      _InternalBuildArgs="/p:DotNetPublishUsingPipelines=true /p:OfficialBuildId=$(BUILD.BUILDNUMBER)"
+    fi
+    if [[ "${{ parameters.includeNonPortable }}" == "true" ]]; then
+      _InternalBuildArgs="$_InternalBuildArgs /p:ArcadeBuildNonPortable=true"
+    fi
+    if [[ "${{ parameters.targetRID }}" != "" ]]; then
+      _InternalBuildArgs="$_InternalBuildArgs /p:TargetRid=${{ parameters.targetRID }}"
+    fi
```

Comment:
I think this logic to detect and rephrase the variables can be switched back to variables by using a job template?

Reply:
I switched it back based on combining the steps template and jobs template into one.